File: C/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

// Function prototypes used inside main
int getLettersCount(string text);
int getSentencesCount(string text);
int getWordsCount(string text);

int main(void)
{
    // Gets the Text input from user
    string text = get_string("Text: ");

    //
    int letters = getLettersCount(text);
    int words = getWordsCount(text);
    int sentences = getSentencesCount(text);

    float l = ((float)letters/words) * 100;
    float s = ((float)sentences/words) * 100;

    // Coleman-Liau index
    // index = 0.0588 * L - 0.296 * S - 15.8
    // L is the average number of letters per 100 words in the text
    // S is the average number of sentences per 100 words in the text.
    int index = (int)round(0.0588 * l - 0.296 * s - 15.8);

    // Prints on the Output what is the Grade level based on the Coleman-Liau index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n",index);
    }

}

// Returns the number o letters in a text
int getLettersCount(string text)
{
    int letters = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

// Returns the number of words in a text
int getWordsCount(string text)
{
    int spaces = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            spaces++;
        }
    }

    return spaces + 1;
}

// Returns the number of senteces in a text
int getSentencesCount(string text)
{
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    return sentences;
}

File: Python/sentimental-readability/readability.py
def get_letters_count(text):
    alpha = list(filter(str.isalpha,text))
    return len(alpha)

def get_words_count(text):
    words = text.split()
    return len(words)

def get_sentences_count(text):
    sentences = list(filter(lambda c: c == '.' or c == '?' or c == '!', text))
    return len(sentences)

if __name__ == "__main__":
    text = input("Text: ")

    letters = get_letters_count(text)
    words = get_words_count(text)
    sentences = get_sentences_count(text)

    l = (letters/words) * 100;
    s = (sentences/words) * 100;

    # Coleman-Liau index
    # index = 0.0588 * L - 0.296 * S - 15.8
    # L is the average number of letters per 100 words in the text
    # S is the average number of sentences per 100 words in the text.
    index = int(round(0.0588 * l - 0.296 * s - 15.8))

    # Prints on the Output what is the Grade level based on the Coleman-Liau index
    if index < 1:
        print("Before Grade 1")
    elif (index >= 16):
        print("Grade 16+")
    else:
        print(f"Grade {index}")
